hashes/src/hex.rs

```diff
@@ -95,8 +95,11 @@ fn chars_to_hex(hi: u8, lo: u8) -> Result<u8, Error> {
     let hih = (hi as char).to_digit(16).ok_or(Error::InvalidChar(hi))?;
     let loh = (lo as char).to_digit(16).ok_or(Error::InvalidChar(lo))?;
 
+    assert!(hih < 16); // 4 bits
+    assert!(loh < 16);
+
     let ret = (hih << 4) + loh;
-    Ok(ret as u8)
+    Ok(ret as u8) // Cast is ok because two hex digits fit into a byte.
```

Comment:
This seems like churn

Reply:
Oh? I thought we are trying to get to a place where there are either no casts or they include a comment as to why they are ok - am I wrong?

Reply:
For sure, but a cast in a 2-line function (whose size is now doubled to add asserts), one of which contains `to_digit(16)`, I think is sufficiently well commented by the code itself.

I wouldn't mind adding the comment without the new asserts, so that `git grep` for ` as ` would show that this has been marked.

Reply:
oh I didn't realise this was the function that added the `debug_assert`s. New comment only it is.